### ISSUE TITLE
Support for Alerts

### ISSUE THREAD

**Issue by solonovamax** [JetBrains/markdown#128]
Support for [Alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) in GFM syntax.

For example:
### Note Alert
> [!note]
> Highlights information that users should take into account, even when skimming.

### Tip Alert
> [!tip]
> Optional information to help a user be more successful.

### Important Alert
> [!important]
> Crucial information necessary for users to succeed.

### Warning Alert
> [!warning]
> Critical content demanding immediate user attention due to potential risks.

### Caution Alert
> [!caution]
> Negative potential consequences of an action.

**Comment by adeherdt-r7** [comment]
Would love to see support for this to help in writing documentation and making sure markdowns render properly on GitHub and GitHub's wiki.

Is there any roadmap or release plan for this feature?

**Comment by domsinclair** [comment]
+1 here for support to be added for this, it's such a useful feature for documents.

**Comment by vinceAmstoutz** [comment]
+1 here for support to be added for this! @JetBrains

**Comment by steveepreston** [comment]
Waiting from [2022](https://github.com/orgs/community/discussions/16925#discussion-4085374). Hope this added in 2025

**Comment by Laxystem** [comment]
Note GitHub added some more:

> [!NOTE]  
> Highlights information that users should take into account, even when skimming.

> [!TIP]
> Optional information to help a user be more successful.

> [!IMPORTANT]  
> Crucial information necessary for users to succeed.

> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.

> [!CAUTION]
> Negative potential consequences of an action.

**Comment by solonovamax** [comment]
@Laxystem I've updated the issue with the new ones

**Comment by zbee** [comment]
+1
There also seems to be [this YouTrack issue](https://youtrack.jetbrains.com/issue/IJPL-92400) for this.

**Comment by sockeye-d** [comment]
Would it be a good idea to convert it to a `<blockquote>` and `<p>` with the class `alert-<whatever text was inside the [!whatever]>`, so

```md
> [!NOTE]
```

would become

```html
<blockquote class="alert-note">
<p>
Note
</p>
<!-- ... -->
</blockquote>
```

and

```md
> [!A-CUSTOM-ALERT]
```

would become

```html
<blockquote class="alert-a-custom-alert">
<p>
A custom alert
</p>
<!-- ... -->
</blockquote>
```

And then colors and such can be assigned in CSS according to the class. I think there's already precedent for this, like how codeblocks are given a class corresponding to the language.

**Comment by zbee** [comment]
No, that wouldn't help anything. GitHub wouldn't support that and this is requesting support for GitHub Markdown in JB IDEs - markdown which already renders/compiles to something like that:
<img width="703" height="238" alt="Image" src="https://github.com/user-attachments/assets/e2398b6f-2a83-46ad-9880-965f62f48885" />

**Comment by sockeye-d** [comment]
Ah.

> This is requesting support for GitHub Markdown in JB IDEs

This isn't true, this is a general-purpose markdown library for Kotlin

**Comment by solonovamax** [comment]
> Would it be a good idea to convert it to a `<blockquote>` and `<p>` with the class `alert-<whatever text was inside the [!whatever]>`, so
> 
> ```md
> > [!NOTE]
> ```
> 
> would become
> 
> ```html
> <blockquote class="alert-note">
> <p>
> Note
> </p>
> 
> </blockquote>
> ```
> 
> and
> 
> ```md
> > [!A-CUSTOM-ALERT]
> ```
> 
> would become
> 
> ```html
> <blockquote class="alert-a-custom-alert">
> <p>
> A custom alert
> </p>
> 
> </blockquote>
> ```
> 
> And then colors and such can be assigned in CSS according to the class. I think there's already precedent for this, like how codeblocks are given a class corresponding to the language.

not sure if `<blockquote>` is the best, as semantically they aren't really block quotes, but yeah the idea would be that it would emit html similar to that whether the tag used is `<blockquote>`, `<div>`, or something else.

**Comment by solonovamax** [comment]
> No, that wouldn't help anything. GitHub wouldn't support that and this is requesting support for GitHub Markdown in JB IDEs - markdown which already renders/compiles to something like that:
> <img width="703" height="238" alt="Image" src="https://github.com/user-attachments/assets/e2398b6f-2a83-46ad-9880-965f62f48885" />

this is not exclusively used by jetbrains IDEs.

for example, I was considering using this library for my personal website (before I eventually decided to settle on flexmark-java instead)

iirc I have used this library however in something else, but I'm blanking right now on what.

**Comment by sockeye-d** [comment]
> for example, I was considering using this library for my personal website

I also already use this library for my project [coho](/sockeye-d/coho)

> not sure if \<blockquote\> is the best, as semantically they aren't really block quotes

They're like basically blockquotes if you really think about it

**Comment by Mrbibi38** [comment]
I would love to see this feature! Or at least a solution to achieve something similar.

**Comment by steveepreston** [comment]
Does anyone created an issue in YouTrack too?
seems Jetbrains throws this repo into hell

**Comment by sockeye-d** [comment]
> There also seems to be [this YouTrack issue](https://youtrack.jetbrains.com/issue/IJPL-92400) for this.

Posted [here.](https://github.com/JetBrains/markdown/issues/128#issuecomment-2678210319)

**Comment by steveepreston** [comment]
> Posted https://github.com/JetBrains/markdown/issues/128#issuecomment-2678210319

Time to bump it

**Comment by p3rmiashkin** [comment]
Hi everyone! Good news: GitHub Alerts are [supported now](https://github.com/JetBrains/markdown/pull/189). Soon we'll release the new version of the library with this feature

**Comment by steveepreston** [comment]
Glad hearing that! Thanks!
@p3rmiashkin Please take a look at this too: [[Feature] Support diff code block](https://github.com/JetBrains/markdown/issues/167)

**Comment by p3rmiashkin** [comment]
@steveepreston Please upvote https://youtrack.jetbrains.com/issue/IJPL-246771/Support-diff-syntax-in-Markdown
